### ISSUE TITLE
Extra env/tags/exec options for executors 

### ISSUE THREAD

**Issue by na--** [loadimpact/k6#1300]
Users should be able to inject environment variables and metric tags per-executor, as well as be able to specify a custom function to be executed instead of the `default` one. Something like this:
```js
export let options = {
    execution: {
        someKey: {
            type: "constant-looping-vus",
            vus: 10,
            duration: "60s",

            // Inject/overwrite environment variable key-value pairs
            env: {
                blah: "whatever",
                key1: "value1",
                key2: "value2"
            },

            // Allow users to execute some other *exported* function besides the default one
            exec: "someExportedFunctionThatsNotTheDefault",

            // The ability to inject additional metric tags
            tags: { tag1: "value1", tag2: "value2" },
        },
    },
}
```

These things would allow us to relatively gracefully support multiple scenarios in the same script, a very frequently requested feature.

**Comment by na--** [comment]
My refactoring proposal in https://github.com/loadimpact/k6/issues/1283#issuecomment-568422059, while apparently not very useful for solving that particular issue as I originally meant it to, still seems to me to be a good way both clean the current overly-complicated code and to implement the features I describe above. 

The only missing piece from it is the validation of the `exec` option - we need to validate that `someExportedFunctionThatsNotTheDefault` is actually a valid exported function name before we start executing the script... So, probably a new `Runner` method would be required to get all of the exported function names or something like that.

**Comment by imiric** [comment]
This should technically be done (see my [`feat/1007-1300-env-tags-exec-options` branch](https://github.com/loadimpact/k6/compare/fix/1007-1283-context-cancelled...imiric:feat/1007-1300-env-tags-exec-options)), but I'll hold off on creating a PR for it until after `fix/1007-1283-context-cancelled` (#1368) is merged, since it's based on it.

**Comment by na--** [comment]
Closed by https://github.com/loadimpact/k6/pull/1428
